src/MSLGenerator.cpp

```diff
@@ -482,33 +485,32 @@ bool MSLGenerator::Generate(HLSLTree* tree, Target target, const char* entryName
     }
 
     if (m_tree->NeedsFunction("mul")) {
+        const char* am = (flags & Flag_PackMatrixRowMajor) ? "m * a" : "a * m";
+        const char* ma = (flags & Flag_PackMatrixRowMajor) ? "a * m" : "m * a";
```

Comment:
The implications of row/column major order are much greater than simply the order of the operands in mul. The implementation of operator [] would also have to change, as well as the _ij member accessors. *But* this is a good start!